static/locate_mirror.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/20.04_testing/lib.sh)
+
+root_check
+
+# Check where the best mirrors are and update
+msg_box "To make downloads as fast as possible when updating you should have mirrors that are as close to you as possible.
+This VM comes with mirrors based on servers in that where used when the VM was released and packaged.
+
+If you are located outside of Europe, we recomend you to change the mirrors so that downloads are faster."
+print_text_in_color "$ICyan" "Checking current mirror..."
+print_text_in_color "$ICyan" "Your current server repository is: $REPO"
+
+if [[ "no" == $(ask_yes_or_no "Do you want to try to find a better mirror?") ]]
+then
+    print_text_in_color "$ICyan" "Keeping $REPO as mirror..."
+    sleep 1
+else
+    if [[ "$KEYBOARD_LAYOUT" =~ ,|/|_ ]]
+    then
+        msg_box "Your keymap contains more than one language, or a special character. ($KEYBOARD_LAYOUT)\nThis script can only handle one keymap at the time.\nWe will now exit."
+        exit 1
```

Comment:
Maybe a small heads up: would that cancel the whole startup script if it returns a 1?
I think the startup script should continue with the default mirror in that case, instead.

Reply:
The script will exit, and don't do any changes to the mirror, keeping the one that were there from the beginning.